packages/sdk-v2/src/sdk/savings/sdk.ts

```diff
@@ -0,0 +1,74 @@
+import React from 'react'
+import { ethers } from "ethers";
+import { BaseSDK } from '../base/sdk'
+import { getSigner } from '../base/react'
+
+
+export class SavingsSDK extends BaseSDK {
+
+  //simple helper for checking an active savings account
+  async hasBalance(
+    account:string
+  ): Promise<boolean | undefined> {
+    const contract = this.getContract("GoodDollarStaking")
+    if (contract && account) {
+      const balance = (await contract.balanceOf(account)).toString()
+      const hasBalance = balance !== '0'
+      return hasBalance
+    }
+  }
+
+  async onTokenTransfer( 
+    account: string, 
+    amount: string, 
+    onSent?: (transactionHash: string) => void): Promise<void | Error> {
+
+    const contract = this.getContract("GoodDollar") 
+    const stakeContract = this.getContract("GoodDollarStaking") 
+    try {
+      const signer = await getSigner(this.signer, account)
```

Comment:
i dont understand this `getSigner`
if this.signer === undefined then you dont do anything
otherwise you simply use this.signer
also the input `account` looks redundant to me. the signer is the account. there's no need to check if signer.address===account? how is that even possible.

Reply:
I have to dive into this more tomorrow

Reply:
was unnecessary so removed